cpp/tests/io/parquet_test.cpp

```diff
@@ -1105,12 +1105,32 @@ TEST_F(ParquetWriterTest, HostBuffer)
     cudf::io::parquet_writer_options::builder(cudf::io::sink_info(&out_buffer), expected)
       .metadata(&expected_metadata);
   cudf::io::write_parquet(out_opts);
-  cudf::io::parquet_reader_options in_opts = cudf::io::parquet_reader_options::builder(
-    cudf::io::source_info(out_buffer.data(), out_buffer.size()));
-  const auto result = cudf::io::read_parquet(in_opts);
 
-  CUDF_TEST_EXPECT_TABLES_EQUAL(expected, result.tbl->view());
-  cudf::test::expect_metadata_equal(expected_metadata, result.metadata);
+  // host buffer
+  {
+    cudf::io::parquet_reader_options in_opts = cudf::io::parquet_reader_options::builder(
+      cudf::io::source_info(out_buffer.data(), out_buffer.size()));
+    const auto result = cudf::io::read_parquet(in_opts);
+
+    CUDF_TEST_EXPECT_TABLES_EQUAL(expected, result.tbl->view());
+    cudf::test::expect_metadata_equal(expected_metadata, result.metadata);
+  }
+
+  // device buffer
+  {
+    auto const d_input = cudf::detail::make_device_uvector_sync(
+      cudf::host_span<uint8_t const>{reinterpret_cast<uint8_t const*>(out_buffer.data()),
+                                     out_buffer.size()},
+      cudf::get_default_stream());
+    auto const d_buffer = cudf::device_span<std::byte const>(
```

Comment:
Might not be required
```suggestion
    auto const d_buffer = cudf::device_span(
```

Reply:
Still needed due to the CTAD failure:
```bash
/home/yunsongw/dev/rapids/cudf/cpp/tests/io/parquet_test.cpp:1126:73: error: class template argument deduction failed:
 1126 |       reinterpret_cast<std::byte const*>(d_input.data()), d_input.size());
      |                                                                         ^

```